core/google/api/core/page_iterator.py

```diff
@@ -291,7 +293,8 @@ class HTTPIterator(Iterator):
     def __init__(self, client, api_request, path, item_to_value,
                  items_key=_DEFAULT_ITEMS_KEY,
                  page_token=None, max_results=None, extra_params=None,
-                 page_start=_do_nothing_page_start, next_token=_NEXT_TOKEN):
+                 page_start=_do_nothing_page_start, next_token=_NEXT_TOKEN,
+                 retry=None):
```

Comment:
I wouldn't make this change. Since `api_request` is a callable, you can wrap it in retry before passing to `page_iterator`, e.g.:

```python
api_request = functools.partial(self._call_api, retry)
```

Reply:
Done.